dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/DependencyInjectionTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Linq.Expressions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.VectorData;
+using Xunit;
+
+namespace VectorDataSpecificationTests;
+
+public abstract class DependencyInjectionTests<TVectorStore, TCollection, TKey, TRecord>
+    where TVectorStore : VectorStore
+    where TCollection : VectorStoreCollection<TKey, TRecord>
+    where TKey : notnull
+    where TRecord : class
+{
+    protected abstract void PopulateConfiguration(ConfigurationManager configuration, object? serviceKey = null);
+
+    protected abstract void RegisterVectorStore(IServiceCollection services, ServiceLifetime lifetime, object? serviceKey = null);
+
+    protected abstract void RegisterCollection(IServiceCollection services, ServiceLifetime lifetime, string collectionName = "name", object? serviceKey = null);
+
+    [Fact]
+    public void ServiceCollectionCantBeNull()
+    {
+        Assert.Throws<ArgumentNullException>(() => this.RegisterVectorStore(null!, ServiceLifetime.Singleton, serviceKey: null));
+        Assert.Throws<ArgumentNullException>(() => this.RegisterVectorStore(null!, ServiceLifetime.Singleton, serviceKey: "notNull"));
+        Assert.Throws<ArgumentNullException>(() => this.RegisterCollection(null!, ServiceLifetime.Singleton, serviceKey: null));
+        Assert.Throws<ArgumentNullException>(() => this.RegisterCollection(null!, ServiceLifetime.Singleton, serviceKey: "notNull"));
+    }
+
+    [Fact]
+    public void CollectionNameCantBeNullOrEmpty()
+    {
+        HostApplicationBuilder builder = this.CreateHostBuilder();
```

Comment:
Do we need Hosting here? I'm used to doing DI tests simply via service collections, no host builder or anything.